sampling: Handle `ErrNOSPC` and `os.PathError` (#8407)
elastic/apm-server: changelogs/head.asciidoc

[float]
==== Bug fixes
- Fix race for deducing destination service fields for OTel bridge {pull}8363[8363]

> - Limit the the local badger database space to 3GB by default for tail-based sampling {pull}8407[8407]

Reviewer: ```suggestion
When tail-based sampling is enabled, limit the local disk space for the feature to 3GB by default {pull}8407[8407]
```
